actions/github-release-dependency/main.go goes into panic if more than 1000 releases

actions/github-release-dependency/main.go can into `panic` if the repo we're listing the releases from has more than 1000 of them; for example:

```
panic: unable to list existing releases for SAP/SapMachine
	GET https://api.github.com/repos/SAP/SapMachine/releases?page=11&per_page=100: 422 Only the first 1000 results are available. []

goroutine 1 [running]:
main.main()
	/Users/anthony/workspaces/paketo-buildpacks/pipeline-builder/actions/github-release-dependency/main.go:90 +0xbb0
```

## Expected Behavior
if GH supported after the 1000th listed release, it would be just fine

## Current Behavior
after the 1000th release will throw 422

## Possible Solution
Unfortunately, [while the UI has a `pre_release` filter](https://docs.github.com/en/repositories/releasing-projects-on-github/searching-a-repositorys-releases#search-syntax-for-searching-releases-in-a-repository), t[here's not such a filter for the API](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28) - I suggest we stop `panic`ing at that line https://github.com/paketo-buildpacks/pipeline-builder/blob/main/actions/github-release-dependency/main.go#L90 OR stop searching after the 10th page , because there's no way the buildpacks would need such old releases


## Motivations
SAPMachine has more than 1000 releases, most of them as `prerelease`s - still, that's more than 1000 and it breaks the `pipelinebuilder` action to update the SAP Machine JDKs

--- a/actions/github-release-dependency/main.go
+++ b/actions/github-release-dependency/main.go
@@ -107,7 +107,7 @@ func main() {
 			}
 		}
 
-		if rsp.NextPage == 0 {
+		if rsp.NextPage == 0 || rsp.NextPage > 10 {
 			break
 		}
 		opt.Page = rsp.NextPage
@@ -136,10 +136,9 @@ func main() {
 		panic(fmt.Errorf("unable to get latest version\n%w", err))
 	}
 	latestSource := actions.Outputs{}
-	if len(sources) != 0{
+	if len(sources) != 0 {
 		latestSource["source"] = sources[latestVersion.Original()]
 	}
-
 
 	url := versions[latestVersion.Original()]
 	outputs, err := actions.NewOutputs(url, latestVersion, latestSource)

